pyvrp/cpp/search/LocalSearch.cpp

```diff
@@ -390,20 +409,64 @@ void LocalSearch::loadSolution(Solution const &solution)
         vehicleOffset[vehType] = vehicleOffset[vehType - 1] + prevAvail;
     }
 
-    // Load routes from solution.
-    for (auto const &solRoute : solution.routes())
+    std::vector<bool> isNewRoute(solution.numRoutes(), true);
+    std::vector<bool> keepExistingRoute(routes.size(), false);
+    auto const &solRoutes = solution.routes();
+
+    // Identify LS routes that are also in the given solution.
+    for (size_t idx = 0; idx != solution.numRoutes(); ++idx)
     {
-        // Determine index of next route of this type to load, where we rely
-        // on solution to be valid to not exceed the number of vehicles per
-        // vehicle type.
-        auto const r = vehicleOffset[solRoute.vehicleType()]++;
-        Route &route = routes[r];
+        auto const &solRoute = solRoutes[idx];
+        auto const vehicleType = solRoute.vehicleType();
+
+        auto const start = routes.begin() + vehicleOffset[vehicleType];
+        auto const end = vehicleType + 1 == data.numVehicleTypes()
+                             ? routes.end()
+                             : routes.begin() + vehicleOffset[vehicleType + 1];
+
+        auto const equal = [&](auto const &route) { return route == solRoute; };
+        auto const route = std::find_if(start, end, equal);
+        if (route != end)
+        {
+            isNewRoute[idx] = false;
+            keepExistingRoute[std::distance(routes.begin(), route)] = true;
+        }
+    }
```

Comment:
What worries me is that this loop is quadratic in the number of vehicles. The other loops are all linear. Can we somehow make this faster too?